en/services/parameter.md

```diff
@@ -7,17 +7,26 @@ The parameter protocol is used to exchange key system settings and guarantees de
 
 It can be both implemented on a microcontroller (e.g. the pxIMU with ARM7) and in standard software (e.g. px_multitracker process in Linux).
 
-## Supported Data Types
+## Message/Enum Summary
 
-MAVLink (v1.0, v2.0) supports these data types:
+Message | Description
+-- | --
+<span id="PARAM_REQUEST_LIST"></span>[PARAM_REQUEST_LIST](../messages/common.md#PARAM_REQUEST_LIST) | Request all parameters. The recipient broadcast all parameter values using [PARAM_VALUE](#PARAM_VALUE).
+<span id="PARAM_REQUEST_READ"></span>[PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ) | Request a single parameter. The recipient broadcasts the specified parameter value using [PARAM_VALUE](#PARAM_VALUE).
+<span id="PARAM_SET"></span>[PARAM_SET](../messages/common.md#PARAM_SET) | Send command to set a specified parameter to a value. After the value has been set (whether successful or not), the recipient should broadcast the current value using [PARAM_VALUE](#PARAM_VALUE).
+<span id="PARAM_VALUE"></span>[PARAM_VALUE](../messages/common.md#PARAM_VALUE) | The current value of a parameter, broadcast in response to a request to get one or more parameters ([PARAM_REQUEST_READ](#PARAM_REQUEST_READ),[PARAM_REQUEST_LIST](#PARAM_REQUEST_LIST)) or whenever a parameter is set/changes ([PARAM_SET](#PARAM_SET))
 
-* `uint32_t` - 32bit unsigned integer (use the ENUM value [MAV_PARAM_TYPE_UINT32](../messages/common.md#MAV_PARAM_TYPE_UINT32))
-* `int32_t` - 32bit signed integer (use the ENUM value [MAV_PARAM_TYPE_INT32](../messages/common.md#MAV_PARAM_TYPE_INT32))
-* `float` - IEEE754 single precision floating point number (use the ENUM value [MAV_PARAM_TYPE_FLOAT](../messages/common.md#MAV_PARAM_TYPE_FLOAT))
+Enum | Description
+-- | --
+<span id="MAV_PARAM_TYPE"></span>[MAV_PARAM_TYPE](../messages/common.md#MAV_PARAM_TYPE) | The [PARAM_SET](#PARAM_SET) and [PARAM_VALUE](#PARAM_VALUE) store/encode parameter values within a `float` field. This type conveys the real type of the encoded parameter value.
 
-> **Note** All parameters are send as the float value of `mavlink_param_union_t`, which means that a parameter should be byte-wise converted with this union to a byte-wise float (no type conversion). 
-  This is necessary in order to not limit the maximum precision for scaled integer params. 
-  E.g. GPS coordinates can only be expressed with single float precision up to a few meters, while GPS coordinates in 1E7 scaled integers provide very high accuracy.
```

Comment:
@julianoes I deleted the two lines above because it isn't obvious how they are relevant/make sense. Specifically it says "This is necessary in", but it isn't clear what part of the above "This" is. 

What I am guessing is that they are saying that normally you might just cast your integer to a float and send that. However if you do that, large numbers like 1E7 scaled integers can't be send. So a better approach is to do a bytewise conversion - the data gets sent as a float, but the float might look nothing like the original data representation.

Would that be correct?

Reply:
UPshot, the point here is "Don't just do a standard cast", do a bytewise copy and set the type.

Reply:
Yes I think that works.

Reply:
THanks. I'll add a note  - which I hope captures this accurately ...

> **Note** A byte-wise conversion is needed, rather than a simple cast, to enable larger integers to be sent (e.g. 1E7 scaled integers can be useful for encoding some types of data, but lose precision if cast directly to floats).